compiler/optimizations/loopInvariantCodeMotion.cpp

```diff
@@ -117,13 +117,21 @@ class Loop {
       delete bitExits;
     }
 
+    LoopStmt* getLoopAST() const {
+      return loopAST;
+    }
+
     // This function exists to place an expr in the "preheader" of the loop
     void insertBefore(Expr* expr) {
       if (loopAST) {
         loopAST->insertBefore(expr->remove());
       }
     }
 
+    bool isGpuBound() const {
+      return isLoopGpuBound((CForLoop*) loopAST);
```

Comment:
I'd prefer 

```c++
CForLoop* cLoop = toCForLoop(loopAST);
INT_ASSERT(cLoop);
return isLoopGpuBound(cLoop);
```

I may be using `INT_ASSERT` more than the average compiler dev, though.